pulsar-functions/runtime-all/src/main/java/org/apache/pulsar/functions/instance/JavaInstanceMain.java

```diff
@@ -101,7 +101,13 @@ public static void main(String[] args) throws Exception {
                 main.getClass().getDeclaredMethod("start", String[].class, ClassLoader.class, ClassLoader.class);
 
         System.out.println("Starting function instance...");
-        method.invoke(main, args, functionInstanceClsLoader, root);
+        try {
+            method.invoke(main, args, functionInstanceClsLoader, root);
+        } catch (InvocationTargetException e) {
```

Comment:
Why not catch `Throwable`? Have you checked if a uncaught throwable handler for the thread would be useful? The exception might happen earlier in the flow and the uncaught throwable handler would handle that. It's used on the broker side.

Reply:
Great suggestion, I'll take a look.

Reply:
I changed it to catch `Throwable` and increased the scope of the `try` block. I don't believe we need to add a default exception handler though. Also, exceptions earlier in the method will lead to the program exiting. The primary reason it didn't exit in the current method is because the invoked method created another thread.